src/node_api.cc

```diff
@@ -1067,10 +1073,9 @@ napi_status napi_define_properties(napi_env env,
     CHECK_NEW_FROM_UTF8(isolate, name, p->utf8name);
 
     v8::PropertyAttribute attributes =
-        v8impl::V8PropertyAttributesFromAttributes(
-            (napi_property_attributes)(p->attributes & ~napi_static_property));
+        v8impl::V8PropertyAttributesFromAttributes(p->attributes);
 
-    if (p->method) {
+    if (p->method != nullptr) {
```

Comment:
> Slightly off-topic and basic, but are the C `NULL` and the C++ `nullptr` interoperable?

Yup. :) The different names are just historical oddities because C++ `NULL` is not defined the same way as C `NULL`, so C++11 introduced `nullptr` as a direct equivalent to C `NULL`.